packages/constants/assets.ts

```diff
@@ -1,165 +1,148 @@
-import { Asset } from '@penumbra-zone/types';
+import { DenomMetadata } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1alpha1/asset_pb';
+import { base64ToUint8Array } from '@penumbra-zone/types';
 
-export interface AssetPattens {
-  lpNftPattern: RegExp;
-  delegationTokenPattern: RegExp;
-  proposalNftPattern: RegExp;
-  unbondingTokenPattern: RegExp;
-  votingReceiptPattern: RegExp;
+export interface AssetPatterns {
+  lpNft: RegExp;
+  delegationToken: RegExp;
+  proposalNft: RegExp;
+  unbondingToken: RegExp;
+  votingReceipt: RegExp;
 }
 
-export const assetPatterns: AssetPattens = {
-  lpNftPattern: new RegExp('^lpnft_'),
-  delegationTokenPattern: new RegExp('^delegation_'),
-  proposalNftPattern: new RegExp('^proposal_'),
-  unbondingTokenPattern: new RegExp('^unbonding_'),
-  votingReceiptPattern: new RegExp('^voted_on_'),
+export const assetPatterns: AssetPatterns = {
+  lpNft: new RegExp('^lpnft_'),
+  delegationToken: new RegExp('^delegation_'),
+  proposalNft: new RegExp('^proposal_'),
+  unbondingToken: new RegExp('^unbonding_'),
+  votingReceipt: new RegExp('^voted_on_'),
 };
 
-// TODO: This should only have things like icon or other fields not available in assets proto
-export const assets: Asset[] = [
+export interface Asset {
+  metadata: DenomMetadata;
+  icon?: string;
+}
+
+export const AssetRegistry: Asset[] = [
```

Comment:
Not at all a blocker, but what do you think of making this an object, keyed by e.g., display name? Then you could just look up metadata on the object by name, rather than iterating over an array and memoizing the result like you're doing [here](https://github.com/penumbra-zone/web/pull/419/files#diff-4ec70347e05c7bcd69a141c28ebc60dfbf558278548fe0b51ce29b002ff07188R7).

Reply:
note there are about a thousand assets that will eventually be in this table